dask/bag/core.py

```diff
@@ -1117,23 +1118,40 @@ def repartition(self, npartitions):
         --------
         >>> b.repartition(5)  # set to have 5 partitions  # doctest: +SKIP
         """
-        if npartitions > self.npartitions:
-            msg = ("Repartition only supports going to fewer partitions\n"
-                   " old: %d  new: %d")
-            raise NotImplementedError(msg % (self.npartitions, npartitions))
-        npartitions_ratio = self.npartitions / npartitions
-        new_partitions_boundaries = [int(old_partition_index * npartitions_ratio)
-                                     for old_partition_index in range(npartitions + 1)]
         new_name = 'repartition-%d-%s' % (npartitions, tokenize(self))
+        if npartitions == self.npartitions:
+            return self
+        elif npartitions < self.npartitions:
+            ratio = self.npartitions / npartitions
+            new_partitions_boundaries = [int(old_partition_index * ratio)
+                                         for old_partition_index in range(npartitions + 1)]
+
+            dsk = {}
+            for new_partition_index in range(npartitions):
+                value = (list, (toolz.concat,
+                                [(self.name, old_partition_index)
+                                 for old_partition_index in
+                                 range(new_partitions_boundaries[new_partition_index],
+                                       new_partitions_boundaries[new_partition_index + 1])]))
+                dsk[new_name, new_partition_index] = value
+        else:  # npartitions > self.npartitions
+            ratio = npartitions / self.npartitions
+            split_name = 'split-%s' % tokenize(self)
```

Comment:
Since the tasks here rely on `npartitions`, this should be in the key as well, either in the call to `tokenize`, or just as a string as done for `new_name` above.